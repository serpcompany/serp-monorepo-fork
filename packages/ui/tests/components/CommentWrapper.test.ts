import { mockNuxtImport } from '@nuxt/test-utils/runtime';
import { afterAll, describe, expect, it, vi } from 'vitest';
import { ref } from 'vue';
// Import mocks first so they're available when components are imported
import { cleanupDateMocks } from '../mockDateImports';
import '../mockUseUserSession';
// Then import components
import CommentWrapper from '../../components/CommentWrapper.vue';
import ComponentRender from '../componentRender';

// Mock composables used in CommentWrapper.vue
mockNuxtImport('useToast', () => () => ({
  add: vi.fn()
}));

mockNuxtImport('useFetch', () => async () => ({
  data: { value: { message: 'success' } },
  error: { value: null }
}));

mockNuxtImport('useRequestHeaders', () => () => ({}));

describe('CommentWrapper Snapshot', () => {
  const baseProps = {
    module: 'test-module',
    id: 1,
    currentIndex: 0,
    initialMessageLimit: '10',
    maxLineLimit: '40',
    maxShowingDepth: '5',
    maxCommentLength: '1000',
    depthLength: 0,
    commentBackgroundColor: 'white',
    commentTextColor: '#1d2129',
    userNameColor: 'rgb(6, 177, 183)',
    wrapperSize: '1024',
    parentIds: [],
    parentIndices: [],
    comment: {
      id: 101,
      image: 'https://example.com/avatar.jpg',
      name: 'Test User',
      content: 'This is a test comment.\nIt has two lines.',
      timestamp: '1973-04-29T21:33:09Z',
      createdAt: '1973-04-29T21:33:09+00:00',
      updatedAt: '1973-04-29T21:33:09+00:00',
      replies: [],
      replyCount: 0,
      email: 'test@example.com',
      lineCount: 2
    }
  };
  it.each([
    ['default comment', { props: baseProps }],
    [
      'updated comment (with Updated badge)',
      {
        props: {
          ...baseProps,
          comment: {
            ...baseProps.comment,
            updatedAt: '1974-04-29T21:33:09+00:00'
          }
        }
      }
    ],
    [
      'comment with replies',
      {
        props: {
          ...baseProps,
          comment: {
            ...baseProps.comment,
            replies: [
              {
                id: 201,
                image: 'https://example.com/reply-avatar.jpg',
                name: 'Reply User',
                content: 'This is a reply.',
                timestamp: '1973-04-29T21:33:09Z',
                createdAt: '1973-04-29T21:33:09+00:00',
                updatedAt: '1973-04-29T21:33:09+00:00',
                replies: [],
                replyCount: 0,
                email: 'reply@example.com',
                lineCount: 1
              }
            ],
            replyCount: 1
          }
        }
      }
    ],
    [
      'comment with nested replies',
      {
        props: {
          ...baseProps,
          comment: {
            ...baseProps.comment,
            replies: [
              {
                id: 202,
                image: 'https://example.com/reply-avatar.jpg',
                name: 'Reply User',
                content: 'This is a reply.',
                timestamp: '1973-04-29T21:33:09Z',
                createdAt: '1973-04-29T21:33:09+00:00',
                updatedAt: '1973-04-29T21:33:09+00:00',
                replies: [
                  {
                    id: 203,
                    image: 'https://example.com/reply-avatar.jpg',
                    name: 'Reply User',
                    content: 'This is a reply.',
                    timestamp: '1973-04-29T21:33:09Z',
                    createdAt: '1973-04-29T21:33:09+00:00',
                    updatedAt: '1973-04-29T21:33:09+00:00',
                    replies: [],
                    replyCount: 0,
                    email: 'reply@example.com',
                    lineCount: 1
                  },
                  {
                    id: 204,
                    image: 'https://example.com/reply-avatar.jpg',
                    name: 'Reply User',
                    content: 'This is a reply.',
                    timestamp: '1973-04-29T21:33:09Z',
                    createdAt: '1973-04-29T21:33:09+00:00',
                    updatedAt: '1973-04-29T21:33:09+00:00',
                    replies: [],
                    replyCount: 0,
                    email: 'reply@example.com',
                    lineCount: 1
                  }
                ],
                replyCount: 2,
                email: 'reply@example.com',
                lineCount: 1
              }
            ],
            replyCount: 1
          }
        }
      }
    ]
  ])(
    'renders %s correctly',
    async (desc: string, options: { props: unknown }) => {
      const html = await ComponentRender(
        `CommentWrapper ${desc}`,
        options,
        CommentWrapper
      );
      expect(html).toMatchSnapshot();
    }
  );

  afterAll(() => {
    cleanupDateMocks();
  });
});
