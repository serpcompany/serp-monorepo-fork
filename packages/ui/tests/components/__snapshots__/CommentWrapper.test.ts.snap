// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CommentWrapper Snapshot > renders comment with nested replies correctly 1`] = `
"<div data-v-d830712d="" class="comment-wrapper">
  <div data-v-d830712d="" class="wrapper">
    <div data-v-d830712d="" class="w-12 rounded-t-full"><img data-v-d830712d="" alt="Avatar" data-nuxt-img="" srcset="https://example.com/avatar.jpg 1x, https://example.com/avatar.jpg 2x" class="h-12 w-12 rounded-full p-1" src="https://example.com/avatar.jpg"></div>
    <div data-v-d830712d="" class="comment-wrapper">
      <div data-v-d830712d="" class="name-wrapper">
        <div data-v-d830712d="" class="name text-(--ui-primary)">Test User</div><span data-v-d830712d="" class="dot">•</span><span data-v-d830712d="" class="time" title="Thursday, January 1, 1970 at 9:00:01 AM">just now</span>
        <!--v-if-->
        <div data-v-d830712d="" class="expand hover:text-(--ui-primary)" style="display: none;"><span data-v-d830712d="" title="Hide Comment">−</span></div>
      </div>
      <div data-v-d830712d="" class="comment bg-white text-black dark:bg-neutral-800 dark:text-white">This is a test comment.
        It has two lines.</div>
      <div data-v-d830712d="" class="comment-box text-black dark:text-white" style="display: none;">
        <div data-v-d830712d="" class="user-name text-(--ui-primary)">Test User</div><textarea data-v-d830712d="" name="addUpdate" class="add-comment bg-white dark:bg-neutral-800" placeholder="Update message" spellcheck="false" aria-label="Add Comment"></textarea><button data-v-d830712d="" aria-label="Update" class="bg-primary hover:bg-primary-800"><span data-v-d830712d="">Update</span></button>
        <div data-v-d830712d="" class="remaining-letter bg-primary"><span data-v-d830712d="">1000</span></div>
      </div>
      <div data-v-d830712d="" class="reply hover:text-(--ui-primary)" style="display: none;">
        <div data-v-d830712d="">Never mind</div>
      </div>
      <div data-v-d830712d="" class="reply">
        <div data-v-d830712d="" class="hover:text-(--ui-primary)">Reply</div><span data-v-d830712d="" class="dot">•</span>
        <div data-v-d830712d="" class="hover:text-(--ui-primary)">Hide replies</div><span data-v-d830712d="" class="dot">•</span>
        <div data-v-d830712d="" class="hover:text-(--ui-primary)"> Update </div><span data-v-d830712d="" class="dot">•</span>
        <div data-v-d830712d="" class="delete">
          <div data-v-d830712d="" class="delete-text">Delete</div>
          <!--v-if-->
        </div>
      </div>
      <!--v-if-->
      <transition-group-stub data-v-d830712d="" name="fade" tag="div" appear="true" persisted="false" css="true">
        <div data-v-d830712d="" class="comment-wrapper">
          <div data-v-d830712d="" class="wrapper">
            <div data-v-d830712d="" class="w-12 rounded-t-full"><img data-v-d830712d="" alt="Avatar" data-nuxt-img="" srcset="https://example.com/reply-avatar.jpg 1x, https://example.com/reply-avatar.jpg 2x" class="h-12 w-12 rounded-full p-1" src="https://example.com/reply-avatar.jpg"></div>
            <div data-v-d830712d="" class="comment-wrapper">
              <div data-v-d830712d="" class="name-wrapper">
                <div data-v-d830712d="" class="name text-(--ui-primary)">Reply User</div><span data-v-d830712d="" class="dot">•</span><span data-v-d830712d="" class="time" title="Thursday, January 1, 1970 at 9:00:01 AM">just now</span>
                <!--v-if-->
                <div data-v-d830712d="" class="expand hover:text-(--ui-primary)" style="display: none;"><span data-v-d830712d="" title="Hide Comment">−</span></div>
              </div>
              <div data-v-d830712d="" class="comment bg-white text-black dark:bg-neutral-800 dark:text-white">This is a reply.</div>
              <div data-v-d830712d="" class="comment-box text-black dark:text-white" style="display: none;">
                <div data-v-d830712d="" class="user-name text-(--ui-primary)">Reply User</div><textarea data-v-d830712d="" name="addUpdate" class="add-comment bg-white dark:bg-neutral-800" placeholder="Update message" spellcheck="false" aria-label="Add Comment"></textarea><button data-v-d830712d="" aria-label="Update" class="bg-primary hover:bg-primary-800"><span data-v-d830712d="">Update</span></button>
                <div data-v-d830712d="" class="remaining-letter bg-primary"><span data-v-d830712d="">1000</span></div>
              </div>
              <div data-v-d830712d="" class="reply hover:text-(--ui-primary)" style="display: none;">
                <div data-v-d830712d="">Never mind</div>
              </div>
              <div data-v-d830712d="" class="reply">
                <div data-v-d830712d="" class="hover:text-(--ui-primary)">Reply</div><span data-v-d830712d="" class="dot">•</span>
                <div data-v-d830712d="" class="hover:text-(--ui-primary)">Hide replies</div><span data-v-d830712d="" class="dot">•</span>
                <div data-v-d830712d="" class="hover:text-(--ui-primary)"> Update </div><span data-v-d830712d="" class="dot">•</span>
                <div data-v-d830712d="" class="delete">
                  <div data-v-d830712d="" class="delete-text">Delete</div>
                  <!--v-if-->
                </div>
              </div>
              <!--v-if-->
              <transition-group-stub data-v-d830712d="" name="fade" tag="div" appear="true" persisted="false" css="true">
                <div data-v-d830712d="" class="comment-wrapper">
                  <div data-v-d830712d="" class="wrapper">
                    <div data-v-d830712d="" class="w-12 rounded-t-full"><img data-v-d830712d="" alt="Avatar" data-nuxt-img="" srcset="https://example.com/reply-avatar.jpg 1x, https://example.com/reply-avatar.jpg 2x" class="h-12 w-12 rounded-full p-1" src="https://example.com/reply-avatar.jpg"></div>
                    <div data-v-d830712d="" class="comment-wrapper">
                      <div data-v-d830712d="" class="name-wrapper">
                        <div data-v-d830712d="" class="name text-(--ui-primary)">Reply User</div><span data-v-d830712d="" class="dot">•</span><span data-v-d830712d="" class="time" title="Thursday, January 1, 1970 at 9:00:01 AM">just now</span>
                        <!--v-if-->
                        <div data-v-d830712d="" class="expand hover:text-(--ui-primary)" style="display: none;"><span data-v-d830712d="" title="Hide Comment">−</span></div>
                      </div>
                      <div data-v-d830712d="" class="comment bg-white text-black dark:bg-neutral-800 dark:text-white">This is a reply.</div>
                      <div data-v-d830712d="" class="comment-box text-black dark:text-white" style="display: none;">
                        <div data-v-d830712d="" class="user-name text-(--ui-primary)">Reply User</div><textarea data-v-d830712d="" name="addUpdate" class="add-comment bg-white dark:bg-neutral-800" placeholder="Update message" spellcheck="false" aria-label="Add Comment"></textarea><button data-v-d830712d="" aria-label="Update" class="bg-primary hover:bg-primary-800"><span data-v-d830712d="">Update</span></button>
                        <div data-v-d830712d="" class="remaining-letter bg-primary"><span data-v-d830712d="">1000</span></div>
                      </div>
                      <div data-v-d830712d="" class="reply hover:text-(--ui-primary)" style="display: none;">
                        <div data-v-d830712d="">Never mind</div>
                      </div>
                      <div data-v-d830712d="" class="reply">
                        <div data-v-d830712d="" class="hover:text-(--ui-primary)">Reply</div>
                        <!--v-if--><span data-v-d830712d="" class="dot">•</span>
                        <div data-v-d830712d="" class="hover:text-(--ui-primary)"> Update </div><span data-v-d830712d="" class="dot">•</span>
                        <div data-v-d830712d="" class="delete">
                          <div data-v-d830712d="" class="delete-text">Delete</div>
                          <!--v-if-->
                        </div>
                      </div>
                      <!--v-if-->
                      <transition-group-stub data-v-d830712d="" name="fade" tag="div" appear="true" persisted="false" css="true"></transition-group-stub>
                      <!--v-if-->
                    </div>
                  </div>
                </div>
                <div data-v-d830712d="" class="comment-wrapper">
                  <div data-v-d830712d="" class="wrapper">
                    <div data-v-d830712d="" class="w-12 rounded-t-full"><img data-v-d830712d="" alt="Avatar" data-nuxt-img="" srcset="https://example.com/reply-avatar.jpg 1x, https://example.com/reply-avatar.jpg 2x" class="h-12 w-12 rounded-full p-1" src="https://example.com/reply-avatar.jpg"></div>
                    <div data-v-d830712d="" class="comment-wrapper">
                      <div data-v-d830712d="" class="name-wrapper">
                        <div data-v-d830712d="" class="name text-(--ui-primary)">Reply User</div><span data-v-d830712d="" class="dot">•</span><span data-v-d830712d="" class="time" title="Thursday, January 1, 1970 at 9:00:01 AM">just now</span>
                        <!--v-if-->
                        <div data-v-d830712d="" class="expand hover:text-(--ui-primary)" style="display: none;"><span data-v-d830712d="" title="Hide Comment">−</span></div>
                      </div>
                      <div data-v-d830712d="" class="comment bg-white text-black dark:bg-neutral-800 dark:text-white">This is a reply.</div>
                      <div data-v-d830712d="" class="comment-box text-black dark:text-white" style="display: none;">
                        <div data-v-d830712d="" class="user-name text-(--ui-primary)">Reply User</div><textarea data-v-d830712d="" name="addUpdate" class="add-comment bg-white dark:bg-neutral-800" placeholder="Update message" spellcheck="false" aria-label="Add Comment"></textarea><button data-v-d830712d="" aria-label="Update" class="bg-primary hover:bg-primary-800"><span data-v-d830712d="">Update</span></button>
                        <div data-v-d830712d="" class="remaining-letter bg-primary"><span data-v-d830712d="">1000</span></div>
                      </div>
                      <div data-v-d830712d="" class="reply hover:text-(--ui-primary)" style="display: none;">
                        <div data-v-d830712d="">Never mind</div>
                      </div>
                      <div data-v-d830712d="" class="reply">
                        <div data-v-d830712d="" class="hover:text-(--ui-primary)">Reply</div>
                        <!--v-if--><span data-v-d830712d="" class="dot">•</span>
                        <div data-v-d830712d="" class="hover:text-(--ui-primary)"> Update </div><span data-v-d830712d="" class="dot">•</span>
                        <div data-v-d830712d="" class="delete">
                          <div data-v-d830712d="" class="delete-text">Delete</div>
                          <!--v-if-->
                        </div>
                      </div>
                      <!--v-if-->
                      <transition-group-stub data-v-d830712d="" name="fade" tag="div" appear="true" persisted="false" css="true"></transition-group-stub>
                      <!--v-if-->
                    </div>
                  </div>
                </div>
              </transition-group-stub>
              <!--v-if-->
            </div>
          </div>
        </div>
      </transition-group-stub>
      <!--v-if-->
    </div>
  </div>
</div>"
`;

exports[`CommentWrapper Snapshot > renders comment with replies correctly 1`] = `
"<div data-v-d830712d="" class="comment-wrapper">
  <div data-v-d830712d="" class="wrapper">
    <div data-v-d830712d="" class="w-12 rounded-t-full"><img data-v-d830712d="" alt="Avatar" data-nuxt-img="" srcset="https://example.com/avatar.jpg 1x, https://example.com/avatar.jpg 2x" class="h-12 w-12 rounded-full p-1" src="https://example.com/avatar.jpg"></div>
    <div data-v-d830712d="" class="comment-wrapper">
      <div data-v-d830712d="" class="name-wrapper">
        <div data-v-d830712d="" class="name text-(--ui-primary)">Test User</div><span data-v-d830712d="" class="dot">•</span><span data-v-d830712d="" class="time" title="Thursday, January 1, 1970 at 9:00:01 AM">just now</span>
        <!--v-if-->
        <div data-v-d830712d="" class="expand hover:text-(--ui-primary)" style="display: none;"><span data-v-d830712d="" title="Hide Comment">−</span></div>
      </div>
      <div data-v-d830712d="" class="comment bg-white text-black dark:bg-neutral-800 dark:text-white">This is a test comment.
        It has two lines.</div>
      <div data-v-d830712d="" class="comment-box text-black dark:text-white" style="display: none;">
        <div data-v-d830712d="" class="user-name text-(--ui-primary)">Test User</div><textarea data-v-d830712d="" name="addUpdate" class="add-comment bg-white dark:bg-neutral-800" placeholder="Update message" spellcheck="false" aria-label="Add Comment"></textarea><button data-v-d830712d="" aria-label="Update" class="bg-primary hover:bg-primary-800"><span data-v-d830712d="">Update</span></button>
        <div data-v-d830712d="" class="remaining-letter bg-primary"><span data-v-d830712d="">1000</span></div>
      </div>
      <div data-v-d830712d="" class="reply hover:text-(--ui-primary)" style="display: none;">
        <div data-v-d830712d="">Never mind</div>
      </div>
      <div data-v-d830712d="" class="reply">
        <div data-v-d830712d="" class="hover:text-(--ui-primary)">Reply</div><span data-v-d830712d="" class="dot">•</span>
        <div data-v-d830712d="" class="hover:text-(--ui-primary)">Hide replies</div><span data-v-d830712d="" class="dot">•</span>
        <div data-v-d830712d="" class="hover:text-(--ui-primary)"> Update </div><span data-v-d830712d="" class="dot">•</span>
        <div data-v-d830712d="" class="delete">
          <div data-v-d830712d="" class="delete-text">Delete</div>
          <!--v-if-->
        </div>
      </div>
      <!--v-if-->
      <transition-group-stub data-v-d830712d="" name="fade" tag="div" appear="true" persisted="false" css="true">
        <div data-v-d830712d="" class="comment-wrapper">
          <div data-v-d830712d="" class="wrapper">
            <div data-v-d830712d="" class="w-12 rounded-t-full"><img data-v-d830712d="" alt="Avatar" data-nuxt-img="" srcset="https://example.com/reply-avatar.jpg 1x, https://example.com/reply-avatar.jpg 2x" class="h-12 w-12 rounded-full p-1" src="https://example.com/reply-avatar.jpg"></div>
            <div data-v-d830712d="" class="comment-wrapper">
              <div data-v-d830712d="" class="name-wrapper">
                <div data-v-d830712d="" class="name text-(--ui-primary)">Reply User</div><span data-v-d830712d="" class="dot">•</span><span data-v-d830712d="" class="time" title="Thursday, January 1, 1970 at 9:00:01 AM">just now</span>
                <!--v-if-->
                <div data-v-d830712d="" class="expand hover:text-(--ui-primary)" style="display: none;"><span data-v-d830712d="" title="Hide Comment">−</span></div>
              </div>
              <div data-v-d830712d="" class="comment bg-white text-black dark:bg-neutral-800 dark:text-white">This is a reply.</div>
              <div data-v-d830712d="" class="comment-box text-black dark:text-white" style="display: none;">
                <div data-v-d830712d="" class="user-name text-(--ui-primary)">Reply User</div><textarea data-v-d830712d="" name="addUpdate" class="add-comment bg-white dark:bg-neutral-800" placeholder="Update message" spellcheck="false" aria-label="Add Comment"></textarea><button data-v-d830712d="" aria-label="Update" class="bg-primary hover:bg-primary-800"><span data-v-d830712d="">Update</span></button>
                <div data-v-d830712d="" class="remaining-letter bg-primary"><span data-v-d830712d="">1000</span></div>
              </div>
              <div data-v-d830712d="" class="reply hover:text-(--ui-primary)" style="display: none;">
                <div data-v-d830712d="">Never mind</div>
              </div>
              <div data-v-d830712d="" class="reply">
                <div data-v-d830712d="" class="hover:text-(--ui-primary)">Reply</div>
                <!--v-if--><span data-v-d830712d="" class="dot">•</span>
                <div data-v-d830712d="" class="hover:text-(--ui-primary)"> Update </div><span data-v-d830712d="" class="dot">•</span>
                <div data-v-d830712d="" class="delete">
                  <div data-v-d830712d="" class="delete-text">Delete</div>
                  <!--v-if-->
                </div>
              </div>
              <!--v-if-->
              <transition-group-stub data-v-d830712d="" name="fade" tag="div" appear="true" persisted="false" css="true"></transition-group-stub>
              <!--v-if-->
            </div>
          </div>
        </div>
      </transition-group-stub>
      <!--v-if-->
    </div>
  </div>
</div>"
`;

exports[`CommentWrapper Snapshot > renders default comment correctly 1`] = `
"<div data-v-d830712d="" class="comment-wrapper">
  <div data-v-d830712d="" class="wrapper">
    <div data-v-d830712d="" class="w-12 rounded-t-full"><img data-v-d830712d="" alt="Avatar" data-nuxt-img="" srcset="https://example.com/avatar.jpg 1x, https://example.com/avatar.jpg 2x" class="h-12 w-12 rounded-full p-1" src="https://example.com/avatar.jpg"></div>
    <div data-v-d830712d="" class="comment-wrapper">
      <div data-v-d830712d="" class="name-wrapper">
        <div data-v-d830712d="" class="name text-(--ui-primary)">Test User</div><span data-v-d830712d="" class="dot">•</span><span data-v-d830712d="" class="time" title="Thursday, January 1, 1970 at 9:00:01 AM">just now</span>
        <!--v-if-->
        <div data-v-d830712d="" class="expand hover:text-(--ui-primary)" style="display: none;"><span data-v-d830712d="" title="Hide Comment">−</span></div>
      </div>
      <div data-v-d830712d="" class="comment bg-white text-black dark:bg-neutral-800 dark:text-white">This is a test comment.
        It has two lines.</div>
      <div data-v-d830712d="" class="comment-box text-black dark:text-white" style="display: none;">
        <div data-v-d830712d="" class="user-name text-(--ui-primary)">Test User</div><textarea data-v-d830712d="" name="addUpdate" class="add-comment bg-white dark:bg-neutral-800" placeholder="Update message" spellcheck="false" aria-label="Add Comment"></textarea><button data-v-d830712d="" aria-label="Update" class="bg-primary hover:bg-primary-800"><span data-v-d830712d="">Update</span></button>
        <div data-v-d830712d="" class="remaining-letter bg-primary"><span data-v-d830712d="">1000</span></div>
      </div>
      <div data-v-d830712d="" class="reply hover:text-(--ui-primary)" style="display: none;">
        <div data-v-d830712d="">Never mind</div>
      </div>
      <div data-v-d830712d="" class="reply">
        <div data-v-d830712d="" class="hover:text-(--ui-primary)">Reply</div>
        <!--v-if--><span data-v-d830712d="" class="dot">•</span>
        <div data-v-d830712d="" class="hover:text-(--ui-primary)"> Update </div><span data-v-d830712d="" class="dot">•</span>
        <div data-v-d830712d="" class="delete">
          <div data-v-d830712d="" class="delete-text">Delete</div>
          <!--v-if-->
        </div>
      </div>
      <!--v-if-->
      <transition-group-stub data-v-d830712d="" name="fade" tag="div" appear="true" persisted="false" css="true"></transition-group-stub>
      <!--v-if-->
    </div>
  </div>
</div>"
`;

exports[`CommentWrapper Snapshot > renders updated comment (with Updated badge) correctly 1`] = `
"<div data-v-d830712d="" class="comment-wrapper">
  <div data-v-d830712d="" class="wrapper">
    <div data-v-d830712d="" class="w-12 rounded-t-full"><img data-v-d830712d="" alt="Avatar" data-nuxt-img="" srcset="https://example.com/avatar.jpg 1x, https://example.com/avatar.jpg 2x" class="h-12 w-12 rounded-full p-1" src="https://example.com/avatar.jpg"></div>
    <div data-v-d830712d="" class="comment-wrapper">
      <div data-v-d830712d="" class="name-wrapper">
        <div data-v-d830712d="" class="name text-(--ui-primary)">Test User</div><span data-v-d830712d="" class="dot">•</span><span data-v-d830712d="" class="time" title="Thursday, January 1, 1970 at 9:00:01 AM">just now</span>
        <!--v-if-->
        <div data-v-d830712d="" class="expand hover:text-(--ui-primary)" style="display: none;"><span data-v-d830712d="" title="Hide Comment">−</span></div>
      </div>
      <div data-v-d830712d="" class="comment bg-white text-black dark:bg-neutral-800 dark:text-white">This is a test comment.
        It has two lines.</div>
      <div data-v-d830712d="" class="comment-box text-black dark:text-white" style="display: none;">
        <div data-v-d830712d="" class="user-name text-(--ui-primary)">Test User</div><textarea data-v-d830712d="" name="addUpdate" class="add-comment bg-white dark:bg-neutral-800" placeholder="Update message" spellcheck="false" aria-label="Add Comment"></textarea><button data-v-d830712d="" aria-label="Update" class="bg-primary hover:bg-primary-800"><span data-v-d830712d="">Update</span></button>
        <div data-v-d830712d="" class="remaining-letter bg-primary"><span data-v-d830712d="">1000</span></div>
      </div>
      <div data-v-d830712d="" class="reply hover:text-(--ui-primary)" style="display: none;">
        <div data-v-d830712d="">Never mind</div>
      </div>
      <div data-v-d830712d="" class="reply">
        <div data-v-d830712d="" class="hover:text-(--ui-primary)">Reply</div>
        <!--v-if--><span data-v-d830712d="" class="dot">•</span>
        <div data-v-d830712d="" class="hover:text-(--ui-primary)"> Update </div><span data-v-d830712d="" class="dot">•</span>
        <div data-v-d830712d="" class="delete">
          <div data-v-d830712d="" class="delete-text">Delete</div>
          <!--v-if-->
        </div>
      </div>
      <!--v-if-->
      <transition-group-stub data-v-d830712d="" name="fade" tag="div" appear="true" persisted="false" css="true"></transition-group-stub>
      <!--v-if-->
    </div>
  </div>
</div>"
`;
